src/geom/SpanSet.cc

```diff
@@ -0,0 +1,640 @@
+/*
+ * LSST Data Management System
+ * Copyright 2008-2016  AURA/LSST.
+ *
+ * This product includes software developed by the
+ * LSST Project (http://www.lsst.org/).
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <https://www.lsstcorp.org/LegalNotices/>.
+ */
+
+#include "lsst/afw/geom/SpanSet.h"
+#include <algorithm>
+#include <iterator>
+
+namespace {
+    /* These classes are used in the erode operator to quickly calculate the
+     * contents of the shrunken SpanSet
+     */
+struct PrimaryRun {
+        int m, y, xmin, xmax;
+};
+
+bool comparePrimaryRun(PrimaryRun const & first, PrimaryRun const & second) {
+        if (first.y != second.y) {
+            return first.y < second.y;
+        } else if (first.m != second.m) {
+            return first.m < second.m;
+        } else {
+            return first.xmin < second.xmin;
+        }
+}
+
+class ComparePrimaryRunY{
+ public:
+            bool operator()(PrimaryRun const& pr, int yval) {
+                return pr.y < yval;
+            }
+            bool operator()(int yval, PrimaryRun const& pr) {
+                return yval < pr.y;
+            }
+};
+
+class ComparePrimaryRunM{
+ public:
+            bool operator()(PrimaryRun const& pr, int mval) {
+                return pr.m < mval;
+            }
+            bool operator()(int mval, PrimaryRun const& pr) {
+                return mval < pr.m;
+            }
+};
+}  // namespace
+
+namespace geom = lsst::afw::geom;
```

Comment:
Actually, we shouldn't need this namespace alias here at all; names in `afw::geom` are already visible.